cypress/e2e/permit2.test.ts

```diff
@@ -0,0 +1,206 @@
+import { MaxUint160, MaxUint256 } from '@uniswap/permit2-sdk'
+
+import { DAI, USDC_MAINNET } from '../../src/constants/tokens'
+
+const APPROVE_BUTTON = '[data-testid="swap-approve-button"]'
+
+/** Initiates a swap and confirms its success. */
+const swaps = () => {
+  // Ensures that the swap button is enabled, since it can be temporarily disabled following approval.
```

Comment:
nit: This comment is iffy because we don't know about the "approval" context from within this helper function. Could you make it more explicit?